app/src/main/java/com/fmsys/snapdrop/MainActivity.java

```diff
@@ -503,4 +504,68 @@ protected void onPostExecute(final String result) {
             }
         }
     }
+
+    public void copyTempToDownloads(JavaScriptInterface.FileHeader fileHeader) {
+        final int notificationId = (int) SystemClock.uptimeMillis();
+
+        //TODO move fileHeader.path to Downloads folder async and handle possible duplciates
```

Comment:
Handling of duplicates was fully done in the previous code, which should IMO be reused. `ContentResolver` / `ContentValues` handles duplicates automatically, while usage of `File` is deprecated as of Android 10 and AFAIK doesn't work at all for external storage anymore. Instead, we should stream the `tempFile` into the `OutputStream` of a modified variant of the old `convertBase64StringToFileAndStoreIt()`. 

https://stackoverflow.com/questions/4741419/is-there-a-better-way-to-write-the-full-contents-of-a-file-to-an-outputstream

What do you think?


https://github.com/fm-sys/snapdrop-android/blob/cb510a8f5ea2354703ce473dcb41a541ec01a3ef/app/src/main/java/com/fmsys/snapdrop/JavaScriptInterface.java#L79-L208

Reply:
> `ContentValues` handles duplicates automatically.

`ContentValues` only handles duplicates on Android 11. It does not work on Android 10.

Reply:
Hey thanks @anggrayudi for chiming in, but SimpleStorage is giving me some troubles, could you take a look at the next commit, maybe you get why the app is crashing if the same file it's send twice...

Reply:
> > `ContentValues` handles duplicates automatically.
> 
> `ContentValues` only handles duplicates on Android 11. It does not work on Android 10.
> 

Are you sure about it? We used it before and we never had any problems (at least I haven't noticed anything). It just adds a "-1" if the filename already exists...

Reply:
Yes, I did tests on Android 10 and 11 @fm-sys . You can try to create a media file in the `Pictures` directory with `ContentValues`. Here's how to reproduce:
1. Use Android 10. Let's say there's a file named `Pictures/profile.png` with media ID 25.
2. Create an image file with `ContentValues` using the same name (`profile`) & mime type (`image/png`), under `Pictures` directory too.
3. A new media file is created with ID 26, but it uses the old file, instead of creating a new file named `profile (1).png`